### PR TITLE
[RFC] Add Pixel "Stuck TBM" simulation workflows to RelVal matrix

### PR DESCRIPTION
**Pull request by mmusich**
Greetings, 
this PR introduces (temporarily) 2018 MC workflows testing the Pixel Stuck TBM simulation (introduced recently in https://github.com/cms-sw/cmssw/pull/25466) in the relval matrix. 
As the feature is switched off by default and the necessary DB conditions are not (yet) available via `autoCond`, the activation flag and the conditions are supplied via customization of the DIGI step.
At the moment only the 2018 (noPU) case is treated as the conditions for the other cases (2017 and PU!=0) are still in the process of being derived.
cc @tsusa

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by fabiocos** [comment]
please test workflow 12424.0,12442.0

**Comment by fabiocos** [comment]
@prebello @zhenhu is this PR ok for you? @kpedro88 it is not upgrade specific, but in case please check it

**Comment by prebello** [comment]
+1

hi @fabiocos  for me is ok

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2894,6 +2894,15 @@ def gen2018HiMix(fragment,howMuch):
                                       '--geometry' : geom
                                       }
 
+    upgradeStepDict['DigiKillPixelStuckTBM'][k] = {'-s':'DIGI:pdigi_valid,L1,DIGI2RAW,HLT:%s'%(hltversion),
+                                                   '--conditions':gt,
+                                                   '--datatier':'GEN-SIM-DIGI-RAW',
+                                                   '-n':'10',
+                                                   '--eventcontent':'FEVTDEBUGHLT',
+                                                   '--geometry' : geom,
+                                                   '--customise_commands':'"process.mix.digitizers.pixel.KillBadFEDChannels = cms.bool(True); process.GlobalTag.toGet = cms.VPSet(cms.PSet(record = cms.string(\'SiPixelStatusScenarioProbabilityRcd\'),tag = cms.string(\'SiPixelQualityProbabilities_2018_noPU_v0_mc\'),connect = cms.string(\'frontier://FrontierPrep/CMS_CONDITIONS\')),cms.PSet(record = cms.string(\'SiPixelStatusScenariosRcd\'),tag = cms.string(\'SiPixelFEDChannelContainer_StuckTBM_2018_v0_mc\'),connect = cms.string(\'frontier://FrontierPrep/CMS_CONDITIONS\')))"'
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: It would be better to put this into an actual customization function somewhere

**Comment by prebello** [comment]
please test workflow 12424.0,12442.0

**Comment by kpedro88** [comment]
@mmusich see https://github.com/mmusich/cmssw/pull/1

**Comment by kpedro88** [comment]
please test workflow 10824.7,10842.7

**Comment by zhenhu** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
+1

**Comment by mmusich** [comment]
@makortel 
Hi Matti, thanks for taking a look. In the meanwhile I realized that the conditions I used specifically for this test are not 100% matching, i.e. there are some non-zero probabilities (in `SiPixelStatusScenarioProbabilityRcd`) assigned to certain scenarios which are not included in the scenario DB object (`SiPixelStatusScenariosRcd`). We'll have to update the condition to make it work correctly, and probably throw in a more graceful way when one attempts to use mismatching conditions.

**Comment by makortel** [comment]
> Hi Matti, thanks for taking a look. In the meanwhile I realized that the conditions I used specifically for this test are not 100% matching, i.e. there are some non-zero probabilities (in `SiPixelStatusScenarioProbabilityRcd`) assigned to certain scenarios which are not included in the scenario DB object (`SiPixelStatusScenariosRcd`). We'll have to update the condition to make it work correctly, and probably throw in a more graceful way when one attempts to use mismatching conditions.

@mmusich Thanks for the quick reply! I'll let you fix the conditions then.

**Pull request merged**